### PR TITLE
Fix for nBlocks<nModules

### PR DESCRIPTION
**Pull request by AdrianoDee**
This realign alpaka and CUDA kernels for  `SiPixelClusterizer` allowing to run with a number of blocks<number of modules.

Tested with the usual `testHLTWithAlpakaPixelReco.sh`

### CODE REVIEW

**Comment by AdrianoDee** [comment]
@fwyzard I've done it here with a simple for loop, do you think it's the way to go? `git diff` is a bit mixed up for some reason.

**Comment by fwyzard** [comment]
Yes, the changes are a bit hard to follow, but overall seem to make sense.
